build.rs

```diff
@@ -1,31 +1,60 @@
-use std::env;
+use std::env::{self, var};
+
+use cmake::Config;
 
 fn main() {
-    //TODO: check cargo features selected
-    //TODO: can conflict/duplicate with make ?
+    let cargo_dir = var("CARGO_MANIFEST_DIR").unwrap();
+    let profile = var("PROFILE").unwrap();
+
+    let target_output_dir = format!("{}/target/{}", cargo_dir, profile);
+    let build_output_dir = format!("{}/build", target_output_dir);
 
     println!("cargo:rerun-if-env-changed=CXXFLAGS");
     println!("cargo:rerun-if-changed=./icicle");
+    println!("cargo:rerun-if-changed=./target/{}", profile); // without this it ignores manual changes to build folder
 
-    let arch_type = env::var("ARCH_TYPE").unwrap_or(String::from("native"));
-    let stream_type = env::var("DEFAULT_STREAM").unwrap_or(String::from("legacy"));
+    let mut cmake = Config::new("./icicle");
+    cmake
+        .define("BUILD_TESTS", "OFF")
+        .out_dir(&target_output_dir)
+        .build_target("icicle");
 
-    let mut arch = String::from("-arch=");
-    arch.push_str(&arch_type);
-    let mut stream = String::from("-default-stream=");
-    stream.push_str(&stream_type);
-
-    let mut nvcc = cc::Build::new();
+    let target_profile: &str;
+    if profile == "release" {
+        target_profile = "Release";
+    } else {
+        target_profile = "Debug";
+    }
```

Comment:
```suggestion
    let target_profile: &str = if profile == "release" {
       "Release"
    } else {
        "Debug"
    };
```

Reply:
I 😄 was sure it was committed as part of the discussion